src/common/actions/repository.js

```diff
@@ -40,7 +48,7 @@ function addRepo(repository) {
   const { id, url } = repository;
 
   return {
-    id,
+    payload: { id },
```

Comment:
It was a bug. Id was not properly nested in the payload, so we didn't know which repo is being added and spinner was not properly shown on 'Add' button.